test/e2e/serial/tests/configuration.go

```diff
@@ -243,67 +243,69 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
-				err = fxt.Client.Update(context.TODO(), nroSchedObj)
-				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(fxt.Client.Update(ctx, nroSchedObj)).To(Succeed())
 			}).WithTimeout(10 * time.Minute).WithPolling(30 * time.Second).Should(Succeed())
 
-			By(fmt.Sprintf("Verify the scheduler object was updated properly with the new scheduler name %q", serialconfig.SchedulerTestName))
+			e2efixture.By("verifying the scheduler object was updated properly with the new scheduler name %q", serialconfig.SchedulerTestName)
 			updatedSchedObj := &nropv1.NUMAResourcesScheduler{}
-			Eventually(func() string {
-				err = fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(nroSchedObj), updatedSchedObj)
-				Expect(err).ToNot(HaveOccurred())
-				return updatedSchedObj.Status.SchedulerName
-			}).WithTimeout(time.Minute).WithPolling(time.Second*15).Should(Equal(serialconfig.SchedulerTestName), "failed to update the schedulerName field,expected %q but found %q", serialconfig.SchedulerTestName, updatedSchedObj.Status.SchedulerName)
+			Eventually(func() bool {
+				Expect(fxt.Client.Get(ctx, client.ObjectKeyFromObject(nroSchedObj), updatedSchedObj)).To(Succeed())
+				return isNROSchedUpToDate(updatedSchedObj)
+			}).WithTimeout(time.Minute).WithPolling(time.Second*15).Should(BeTrue(), "failed to update the schedulerName field")
+			Expect(updatedSchedObj.Status.SchedulerName).To(Equal(serialconfig.SchedulerTestName), "failed to update the schedulerName field,expected %q but found %q", serialconfig.SchedulerTestName, updatedSchedObj.Status.SchedulerName)
 
-			defer func() {
+			defer func(rctx context.Context) {
 				By("reverting the changes under the NUMAResourcesScheduler object")
 				// see https://pkg.go.dev/github.com/onsi/gomega#Eventually category 3
 				Eventually(func(g Gomega) {
 					currentSchedObj := &nropv1.NUMAResourcesScheduler{}
-					err := fxt.Client.Get(context.TODO(), nroSchedKey, currentSchedObj)
-					g.Expect(err).ToNot(HaveOccurred(), "cannot get current %q in the cluster", nroSchedKey.String())
+					g.Expect(fxt.Client.Get(rctx, nroSchedKey, currentSchedObj)).To(Succeed(), "cannot get current %q in the cluster", nroSchedKey.String())
 
 					currentSchedObj.Spec.SchedulerName = initialNroSchedObj.Status.SchedulerName
-					err = fxt.Client.Update(context.TODO(), currentSchedObj)
-					g.Expect(err).ToNot(HaveOccurred())
+					g.Expect(fxt.Client.Update(rctx, currentSchedObj)).To(Succeed())
 				}).WithTimeout(5*time.Minute).WithPolling(10*time.Second).Should(Succeed(), "failed to revert changes the changes to the NRO scheduler object")
 
 				updatedSchedObj := &nropv1.NUMAResourcesScheduler{}
 				Eventually(func() string {
-					err = fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(initialNroSchedObj), updatedSchedObj)
-					Expect(err).ToNot(HaveOccurred())
+					Expect(fxt.Client.Get(rctx, client.ObjectKeyFromObject(initialNroSchedObj), updatedSchedObj)).To(Succeed())
```

Comment:
same here, probably never failed on this because Get is unlikly to fail. I'm seeing a couple of examples in the suite that do this mistake, we should fix in a cleanup PR, I expect that to stabilize the suite more.

Reply:
indeed, fixed all the occurrences in this test.